laravel-live-coding/database/migrations/2023_09_21_3_create_wallets_table.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('wallets', function (Blueprint $table) {
+            $table->id();
+            $table->double('amount')->nullable(false);
+            $table->foreignId('user_id')->constrained('users');
+            $table->foreignId('currency_id')->constrained('currencies');
```

Comment:
**constrained** It is not not enough
we have to write **->constrained('users')->cascadeOnDelete()**
_That means if the **user deleted**, the **row** in wallets with a user **would remove**_
if we need to keep it row, we will use **nullOnDelete()**

Read about cascade one more time,